sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/VectorizedPlainValuesReader.java

```diff
@@ -31,6 +33,8 @@
   private byte[] buffer;
   private int offset;
   private int bitOffset; // Only used for booleans.
+  
+  private final static boolean bigEndianPlatform = ByteOrder.nativeOrder().equals(ByteOrder.BIG_ENDIAN);
```

Comment:
I'm wondering if there is a difference between wrapping the buffer with a ByteBuffer once (in InitFromPage) and if that behaves any differently than reversing for each integer.